apps/backoffice-v2/src/lib/blocks/variants/DefaultBlocks/hooks/useDefaultBlocksLogic/useDefaultBlocksLogic.tsx

```diff
@@ -0,0 +1,396 @@
+import { useParams } from 'react-router-dom';
```

Comment:
This is a new 400 lines file with a lot of logic. Can we think of a way to simplify and shorten this file without compromising usability?